layer/layer_store.go

```diff
@@ -398,10 +398,24 @@ func (ls *layerStore) Get(l ChainID) (Layer, error) {
 	if layer == nil {
 		return nil, ErrLayerDoesNotExist
 	}
+	if !ls.driver.Exists(layer.cacheID) {
+		if err := ls.deleteCorruptLayer(layer); err != nil {
```

Comment:
Even though the layer itself is corrupt, deleting it at this level doesn't really seem right.
For drivers like btrfs this especially wouldn't work (granted it seems like it would be weirder to get into this situation with btrfs).

Reply:
It seems like if we want to recover from this we need a way to repair the layer, but this seems non-trivial from a more generic standpoint.